Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.h

```diff
@@ -57,10 +59,12 @@ namespace Terrain
         AZ_TYPE_INFO(TerrainPhysicsColliderConfig, "{E9EADB8F-C3A5-4B9C-A62D-2DBC86B4CE59}");
         static void Reflect(AZ::ReflectContext* context);
 
-        AZ::Data::AssetId GetDefaultPhysicsAssetId() const;
-
         AZ::Data::Asset<Physics::MaterialAsset> m_defaultMaterialAsset;
         AZStd::vector<TerrainPhysicsSurfaceMaterialMapping> m_surfaceMaterialMappings;
+
+    private:
+        friend class EditorTerrainPhysicsColliderComponent;
+        AZ::Data::AssetId GetDefaultPhysicsMaterialAssetIdEditContext() const;
```

Comment:
suggestion: It might not be worth the hassle of making these types friends and just instead make `GetDefaultPhysicsMaterialAssetIdEditContext` public (especially as it's `const` anyway). The problem with friendship is now everything else in the class (not just this function) is accessible. I realize this is a POD type anyway and things are public by default, but it's not always the case. You can leave as-is but I generally try and avoid `friend` wherever possible, it's usually more trouble than it's worth.

Reply:
`GetDefaultPhysicsMaterialAssetIdEditContext` is only meant to be used for edit context for the asset attribute, if this is made public it looks like users can call it to obtain the default material and that's not something these classes has to offer.

In this particular case the friend class `EditorTerrainPhysicsColliderComponent` is actually doing the edit context reflection for it, so I think it makes sense it can access its properties. Because of this I think it's justified to make it friend.

A different solution would have been to move the edit context reflection to itself, but that's a larger change than required.

Reply:
Using free function to remove the member functions. See comment below. Thanks @hultonha for the suggestion!